src/db.ts

```diff
@@ -289,7 +289,10 @@ export class Db {
    * @param pipeline - An array of aggregation stages to be executed
    * @param options - Optional settings for the command
    */
-  aggregate(pipeline: Document[] = [], options?: AggregateOptions): AggregationCursor {
+  aggregate<T = Document>(
```

Comment:
Should this default to Document? The result of this call could very well be grossly incompatible with an object, and even if it's not it may lead to some confusion since the result is more likely than not to not have all the functionality/expectations of a full document, if my understanding of what the Document type exposes is correct.